test/jdk/java/lang/runtime/CarrierTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @summary Test features provided by the Carrier class.
+ * @compile CarrierTest.java
+ * @run main CarrierTest
+ */
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.runtime.Carrier;
+import java.util.Arrays;
+import java.util.List;
+
+public class CarrierTest {
+    public static void main(String[] args) throws Throwable {
+        primitivesTest();
+        primitivesTestInArrayCarrier();
+        limitsTest();
+    }
+
+    static void assertTrue(boolean test, String message) {
```

Comment:
This test does not use TestNG like the other tests in the package. Is that intentional?

(Please tell me if its weird that I as a total rando pops in and makes comments here!)